__init__.py

```diff
@@ -643,29 +648,38 @@ def _get_speakable_timer_details(self, timers: List[CountdownTimer]) -> str:
 
         return timer_names
 
-    def display_timers(self):
+    def _show_gui(self):
+        """Update the device's display to show the status of active timers.
+
+        Runs once a second via a repeating event to keep the information on the display
+        accurate.
+        """
+        if self.gui.connected:
+            self._update_gui()
+            if self.platform == MARK_II:
+                page = "timer_mark_ii.qml"
+            else:
+                page = "timer_scalable.qml"
+            self.gui.show_page(page, override_idle=True)
+
+    def update_display(self):
         """Update the device's display to show the status of active timers.
 
         Runs once a second via a repeating event to keep the information on the display
         accurate.
         """
         if self.gui.connected:
-            self._display_timers_on_gui()
+            self._update_gui()
         elif self.platform == MARK_I:
             self._display_timers_on_faceplate()
 
-    def _display_timers_on_gui(self):
+    def _update_gui(self):
         """Display active timers on a device that supports the QT GUI framework."""
         timers_to_display = self._select_timers_to_display(display_max=4)
         display_data = [timer.display_data for timer in timers_to_display]
-        if self.platform == MARK_II:
-            page = "timer_mark_ii.qml"
-        else:
-            page = "timer_scalable.qml"
         if timers_to_display:
             self.gui["activeTimers"] = dict(timers=display_data)
             self.gui["activeTimerCount"] = len(timers_to_display)
-            self.gui.show_page(page, override_idle=True)
 
     def _display_timers_on_faceplate(self):
         """Display one timer on a device that supports and Arduino faceplate."""
```

Comment:
In check_for_expired_timers() you call play_wav() directly - shouldn't a message be sent out to the audio service so this doesn't cause device contention? How does this work if somebody else called play_wav() before you?

Reply:
you are right, but I could not find where in the audio service a message is defined to handle playing sounds.  if you know what it is, tell me and I will be happy to change this.  otherwise, I can add a TODO to fix it in the future.

Reply:
Its OK, its a playback control message but I just noticed other skills call this method directly as well. nm